lib/celluloid/io/dns_resolver.rb

```diff
@@ -42,7 +42,10 @@ def resolve(hostname)
 
         query = build_query(hostname)
         @socket.send query.encode, 0, @server.to_s, DNS_PORT
-        data, _ = @socket.recvfrom(MAX_PACKET_SIZE)
+        data, _ = RUBY_VERSION >= "2.3" ?
```

Comment:
I mean, this definitely seems like the wrong place to put this check. In the DNS resolver? It should be using a `Celluloid::IO::UDPSocket`, and these details should be pushed down there.